src/components/text-editor/text-editor.pw.tsx

```diff
@@ -1,1387 +1,790 @@
-  ["Space", "Enter"].forEach((key) =>
-    test(`focuses text input when italic button is deselected using the ${key} key`, async ({
       page,
-      mount,
     }) => {
-      await mount(<TextEditorCustom />);
-
-      const italicButton = page.getByRole("button", { name: "italic" });
-      await italicButton.click();
-
-      await italicButton.press(key);
-
-      const textInput = page.getByRole("textbox");
-      await expect(textInput).toBeFocused();
-    }),
-  );
+      await mount(
+        <TextEditorDefaultComponent
+          value={preformattedValue}
+          onCancel={() => {}}
+        />,
+      );
+      const textbox = await page.locator("div[role='textbox']");
+      await textbox.click();
+      await textbox.pressSequentially(" This is some text", { delay: 100 });
+      const cancelButton = page.locator(
+        "button[data-role='pw-rte-cancel-button']",
+      );
+      await cancelButton.click();
+      const displayedText = await textbox.textContent();
+      expect(displayedText).toBe("Sample text with some formatting applied.");
+    });
+  });
 
-  ["Space", "Enter"].forEach((key) =>
-    test(`focuses text input when bullet list button is deselected using the ${key} key`, async ({
-      page,
+  test.describe("onChange", () => {
+    test("fires when the content of the editor changes", async ({
       mount,
-    }) => {
-      await mount(<TextEditorCustom />);
-
-      const bulletListButton = page.getByRole("button", {
-        name: "bullet-list",
-      });
-      await bulletListButton.click();
-
-      await bulletListButton.press(key);
-
-      const textInput = page.getByRole("textbox");
-      await expect(textInput).toBeFocused();
-    }),
-  );
-
-  ["Space", "Enter"].forEach((key) =>
-    test(`focuses text input when number list button is deselected using the ${key} key`, async ({
       page,
-      mount,
     }) => {
-      await mount(<TextEditorCustom />);
-
-      const numberListButton = page.getByRole("button", {
-        name: "number-list",
-      });
-      await numberListButton.click();
-
-      await numberListButton.press(key);
-
-      const textInput = page.getByRole("textbox");
-      await expect(textInput).toBeFocused();
-    }),
-  );
-
-  test(`should render formatted link`, async ({ mount, page }) => {
-    const linkText = "https://carbon.sage.com";
-    await mount(<TextEditorCustom />);
+      let callbackFired = false;
+      await mount(
+        <TextEditorDefaultComponent
+          onChange={() => {
+            callbackFired = true;
+          }}
+        />,
+      );
+      const textbox = await page.locator("div[role='textbox']");
+      await textbox.click();
+      await textbox.pressSequentially(" This is some text", { delay: 100 });
 
-    const textInput = page.getByRole("textbox");
-    await textInput.fill(linkText);
+      /** Although nasty and discouraged, I've hard-coded a wait here to give
+       * the debounce function time to fire the callback.
```

Comment:
Same thing here too with regards to removing the `I've` from the comment.

```suggestion
      /** Although nasty and discouraged, the hard-coded wait is here to give
       * the debounce function time to fire the callback.
```